spirv_msl.cpp

```diff
@@ -2490,9 +2540,10 @@ bool CompilerMSL::is_member_packable(SPIRType &ib_type, uint32_t index)
 	if (mbr_type.vecsize == 1 || (is_matrix(mbr_type) && mbr_type.vecsize != 3))
 		return false;
 
-	// Only row-major matrices need to be packed.
-	if (is_matrix(mbr_type) && !has_member_decoration(ib_type.self, index, DecorationRowMajor))
-		return false;
+	// Pack if the member's offset doesn't conform to the type's usual
+	// alignment. For example, a float3 at offset 4.
+	if (mbr_offset_curr % get_declared_struct_member_alignment(ib_type, index))
```

Comment:
I do wonder if this needs to come before the check above for matrix type and vecsize != 3. I think it's still possible we hit the case where a mat2 has an offset of 4. Try this declaration:

```
layout(scalar) buffer {
    float a;
    mat2 b; <-- is declared as float2x2, not packed_float2x2
};
```

Reply:
Well, OK, but I had to completely redo support for packed matrices.